signac/core/synced_collections/buffered_collection.py

```diff
@@ -233,4 +199,35 @@ def _flush_buffer(self):
         pass
 
 
-buffer_all = BufferedCollection.buffer_all
+# Have to monkey-patch this into the class after the fact because the
+# flush_all_backends method is not fully defined and bound as a static method
+# until after the class definition is complete.
+BufferedCollection.buffer_all = _CounterFuncContext(  # type: ignore
+    BufferedCollection._flush_all_backends
+)
+"""Enter a globally buffer context for all BufferedCollection instances.
+
+All future operations use the buffer whenever possible. Write operations
+are deferred until the context is exited, at which point all buffered
+backends will flush their buffers. Individual backends may flush their
+buffers within this context if the implementation requires it; this context
+manager represents a promise to buffer whenever possible, but does not
+guarantee that no writes will occur under all circumstances.
+"""
+
+
+# This function provides a more familiar module-scope, function-based interface
+# for enabling buffering rather than calling the class's static method.
+def buffer_all():
+    """Enter a globally buffer context for all BufferedCollection instances.
+
+    All future operations use the buffer whenever possible. Write operations
+    are deferred until the context is exited, at which point all buffered
+    backends will flush their buffers. Individual backends may flush their
+    buffers within this context if the implementation requires it; this context
+    manager represents a promise to buffer whenever possible, but does not
+    guarantee that no writes will occur under all circumstances.
+
+    This method is a trivial alias for :meth:`~.BufferedCollection.buffer_all`
+    """
+    return BufferedCollection.buffer_all
```

Comment:
Do we need the monkey patch? Would this function be enough? (It might need some kind of singleton-like pattern?)

Reply:
I'm not sure I follow your suggestion. If you're just asking whether we can just skip putting `buffer_all` into the namespace of the `BufferedCollection`, we certainly can. From a conceptual standpoint, to me it actually makes more sense in the `BufferedCollection` than as a module-scope function, and I only included the function as a nod to backwards compatibility. If you think that the monkey-patch is confusing, though, then that's fine; we'd just need to preserve the context as a global variable at the module scope and return it via the function.

Reply:
Yes, I think I would prefer only one or the other. Having both the class method and module-level function seems unnecessary. Given that, I would prefer to skip the monkey-patch to put this in the `BufferedCollection` namespace and leave only the module-level function. I don't have strong opinions on this but wanted to at least pose the suggestion because monkey-patches are pretty hacky.

Reply:
That's fine, I'll stick with a global context. I agree that monkey-patches can be hacky and would make things harder to maintain.